lib/puma/cluster/worker.rb

```diff
@@ -126,12 +126,8 @@ def run
 
             while true
               begin
-                b = server.backlog || 0
-                r = server.running || 0
-                t = server.pool_capacity || 0
-                m = server.max_threads || 0
-                rc = server.requests_count || 0
-                payload = %Q!#{base_payload}{ "backlog":#{b}, "running":#{r}, "pool_capacity":#{t}, "max_threads":#{m}, "requests_count":#{rc} }\n!
+                stat_string = Puma::Server::STAT_METHODS.map { |s| "\"#{s}\":#{server.public_send(s) || 0}" }.join(', ')
```

Comment:
Please revert this logic. Calling `Array#map` allocates a new array https://ruby-doc.org/3.3.5/Array.html#method-i-map. And then N String allocations instead of a single string allocation. Calling public send is slower than calling the method directly.

It's not a wild amount of overhead, but I want calling stats to be as low overhead as possible.

Reply:
counterpoint: calling stats typically happens every 30 or 60 seconds, so the performance impact is irrelevant

🎭🪞counterpoint counterpoint: new stats are added very rarely, so the maintainability win of DRYing this is irrelevant

Reply:
reverted